internal/expr/parser.go

```
@@ -265,28 +272,174 @@ func (p *Parser) parseIdentifier() (string, bool) {
+				return fullName{prefix: id, name: idCol}, true, nil
+			}
+		} else {
+			// A column name specified without a table prefix should be in name.
+			return fullName{name: id}, true, nil
+		}
+	}
+
+	cp.restore()
+	return fullName{}, false, nil
+}
+
+func (p *Parser) parseTarget() (fullName, bool, error) {
+	cp := p.save()
+
+	if p.skipByte('&') {
+		return p.parseGoFullName()
+	}
+
+	cp.restore()
+	return fullName{}, false, nil
+}
+
 // parseGoFullName parses a Go type name qualified by a tag name (or asterisk)
-// of the form "TypeName.col_name". On success it returns the parsed fullName,
+// of the form "&TypeName.col_name". On success it returns the parsed FullName,
 // true and nil. If a Go full name is found, but not formatted correctly, false
 // and an error are returned. Otherwise the error is nil.
 func (p *Parser) parseGoFullName() (fullName, bool, error) {
 	cp := p.save()
+
 	if id, ok := p.parseIdentifier(); ok {
-		if p.skipByte('.') {
-			if idField, ok := p.parseIdentifierAsterisk(); ok {
-				return fullName{id, idField}, true, nil
-			}
-			return fullName{}, false,
-				fmt.Errorf("invalid identifier near char %d", p.pos)
+		if !p.skipByte('.') {
+			return fullName{}, false, fmt.Errorf("go object near char %d not qualified", p.pos)
+		}
+
+		idField, ok := p.parseIdentifierAsterisk()
+		if !ok {
+			return fullName{}, false, fmt.Errorf("invalid identifier near char %d", p.pos)
 		}
-		return fullName{}, false,
-			fmt.Errorf("go object near char %d not qualified", p.pos)
+		return fullName{id, idField}, true, nil
 	}
+
 	cp.restore()
 	return fullName{}, false, nil
 }
 
-// parseInputExpression parses an input expression of the form $Type.name.
+// parseList takes a parsing function that returns a fullName and parses a
+// bracketed, comma seperated, list.
+func (p *Parser) parseList(parseFn func(p *Parser) (fullName, bool, error)) ([]fullName, bool, error) {
+	cp := p.save()
+
+	if !p.skipByte('(') {
+		return nil, false, nil
```

Should we return an error here as done elsewhere, replacing the last `nil` with an `fmt.Errorf()` with an relevant error message?

---

When we return false but not an error in the parser its because we've not found the thing we are looking for. If we return an error then we have found it but we thing it is incorrectly formatted.
Here we return `nil, false` if we cant find an opening bracket that signifies the start of the list, i.e. we cant find a list to parse.